kytos/core/interface.py

```diff
@@ -173,19 +193,311 @@ def status_reason(self):
+            fst_a, snd_a = ranges_a[a_i]
+            fst_b, snd_b = ranges_b[b_i]
+            # Moving forward with non-intersection
+            if snd_a < fst_b:
+                a_i += 1
+            elif snd_b < fst_a:
+                b_i += 1
+            else:
+                # Intersection
+                intersection_start = max(fst_a, fst_b)
+                intersection_end = min(snd_a, snd_b)
+                yield [intersection_start, intersection_end]
+                if snd_a < snd_b:
+                    a_i += 1
+                else:
+                    b_i += 1
+
+    @staticmethod
+    def range_difference(
+        ranges_a: list[list[int]],
+        ranges_b: list[list[int]]
+    ) -> list[list[int]]:
+        """The operation is (ranges_a - ranges_b).
+        This method simulates difference of sets. E.g.:
+        [[10, 15]] - [[4, 11], [14, 45]] = [[12, 13]]
+        """
+        result = []
+        a_i, b_i = 0, 0
+        update = True
+        while a_i < len(ranges_a) and b_i < len(ranges_b):
+            if update:
+                start_a, end_a = ranges_a[a_i]
+            else:
+                update = True
+            start_b, end_b = ranges_b[b_i]
+
+            # Moving forward with non-intersection
+            if end_a < start_b:
+                result.append([start_a, end_a])
+                a_i += 1
+            elif end_b < start_a:
+                b_i += 1
+            else:
+                # Intersection
+                if start_a < start_b:
+                    result.append([start_a, start_b - 1])
+
+                if end_a > end_b:
+                    start_a = end_b + 1
+                    update = False
+                    b_i += 1
+                else:
+                    a_i += 1
+
+        # Append last intersection and the rest of ranges_a
+        while a_i < len(ranges_a):
+            if update:
+                start_a, end_a = ranges_a[a_i]
+            else:
+                update = True
+            result.append([start_a, end_a])
+            a_i += 1
+
+        return result
+
+    @staticmethod
+    def can_set_tag_ranges(used_tags, tag_ranges):
+        """Check if all used tags are included in tag_ranges"""
+        tag_n = len(tag_ranges)
+        used_n = len(used_tags)
+        tag_i, used_i = 0, 0
+        while tag_i < tag_n and used_i < used_n:
+            start_tag, end_tag = tag_ranges[tag_i]
+            start_used, end_used = used_tags[used_i]
+
+            if end_tag < start_used:
+                tag_i += 1
+            elif end_used < start_tag:
+                msg = f"Missing all tags from {used_tags[used_i]}"
+                raise KytosSetTagRangeError(msg)
+            else:
+                if start_tag <= start_used and end_tag >= end_used:
+                    tag_i += 1
+                    used_i += 1
+                else:
+                    msg = f"Missing tags from {used_tags[used_i]}"
```

Comment:
I realized that this method is similar to `range_difference`. Now using `self.range_difference(used_tags, tag_ranges)` that returns all the missing tags instead of the first found problem. No need to change unit test, apart from deleting `range_difference` since it is not used anywhere.